app/models/user.rb

```diff
@@ -89,4 +92,16 @@ def default_email_record
     result ||= valid_emails.first
     result
   end
+
+  # Overrides `Devise::Models::Validatable`
```

Comment:
This handles email and password validation, which is what the user authentication concern deals with. So I think it's best to be in the authentication concern instead -- hence the term concern :smile: